infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -196,10 +218,17 @@ private void handleMetadataRequest(@Nonnull RequestMsg request,
 
             LogReplicationSinkManager sinkManager = sessionToSinkManagerMap.get(session);
 
+            // SinkManager not found can indicate a new session discovered or yet to be discovered.
+            // Drop the message if the incoming session is not known to sessionManger, otherwise create a corresponding
```

Comment:
what about the case where this cluster has not discovered that session? we are still not covering that, right?

Reply:
No, thats not covered in this PR. I'll add a TODO about that